include/boost/beast/core/impl/file_win32.ipp

```diff
@@ -137,66 +138,86 @@ open(char const* path, file_mode mode, error_code& ec)
+        desired_access = boost::detail::winapi::GENERIC_READ_;
+
+        share_mode = boost::detail::winapi::FILE_SHARE_READ_;
+
+        creation_disposition = boost::detail::winapi::OPEN_EXISTING_;
+
+        flags_and_attributes = 0x10000000; // FILE_FLAG_RANDOM_ACCESS
         break;
 
     case file_mode::scan:           
-        dw1 = boost::detail::winapi::GENERIC_READ_;
-        dw2 = boost::detail::winapi::OPEN_EXISTING_;
-        dw3 = 0x08000000; // FILE_FLAG_SEQUENTIAL_SCAN
+        desired_access = boost::detail::winapi::GENERIC_READ_;
+
+        share_mode = boost::detail::winapi::FILE_SHARE_READ_;
+
+        creation_disposition = boost::detail::winapi::OPEN_EXISTING_;
+
+        flags_and_attributes = 0x08000000; // FILE_FLAG_SEQUENTIAL_SCAN
         break;
 
     case file_mode::write:          
-        dw1 = boost::detail::winapi::GENERIC_READ_ |
-              boost::detail::winapi::GENERIC_WRITE_;
-        dw2 = boost::detail::winapi::CREATE_ALWAYS_;
-        dw3 = 0x10000000; // FILE_FLAG_RANDOM_ACCESS
+        desired_access = boost::detail::winapi::GENERIC_READ_ |
+                         boost::detail::winapi::GENERIC_WRITE_;
+
+        creation_disposition = boost::detail::winapi::CREATE_ALWAYS_;
+
+        flags_and_attributes = 0x10000000; // FILE_FLAG_RANDOM_ACCESS
         break;
 
     case file_mode::write_new:      
-        dw1 = boost::detail::winapi::GENERIC_READ_ |
-              boost::detail::winapi::GENERIC_WRITE_;
-        dw2 = boost::detail::winapi::CREATE_NEW_;
-        dw3 = 0x10000000; // FILE_FLAG_RANDOM_ACCESS
+        desired_access = boost::detail::winapi::GENERIC_READ_ |
+                         boost::detail::winapi::GENERIC_WRITE_;
+
+        creation_disposition = boost::detail::winapi::CREATE_NEW_;
+
+        flags_and_attributes = 0x10000000; // FILE_FLAG_RANDOM_ACCESS
         break;
 
     case file_mode::write_existing: 
-        dw1 = boost::detail::winapi::GENERIC_READ_ |
-              boost::detail::winapi::GENERIC_WRITE_;
-        dw2 = boost::detail::winapi::OPEN_EXISTING_;
-        dw3 = 0x10000000; // FILE_FLAG_RANDOM_ACCESS
+        desired_access = boost::detail::winapi::GENERIC_READ_ |
+                         boost::detail::winapi::GENERIC_WRITE_;
+
+        creation_disposition = boost::detail::winapi::OPEN_EXISTING_;
+
+        flags_and_attributes = 0x10000000; // FILE_FLAG_RANDOM_ACCESS
         break;
 
     case file_mode::append:         
-        dw1 = boost::detail::winapi::GENERIC_READ_ |
-              boost::detail::winapi::GENERIC_WRITE_;
-        dw2 = boost::detail::winapi::CREATE_ALWAYS_;
-        dw3 = 0x08000000; // FILE_FLAG_SEQUENTIAL_SCAN
+        desired_access = boost::detail::winapi::GENERIC_READ_ |
+                         boost::detail::winapi::GENERIC_WRITE_;
+
+        creation_disposition = boost::detail::winapi::CREATE_ALWAYS_;
+
+        flags_and_attributes = 0x08000000; // FILE_FLAG_SEQUENTIAL_SCAN
```

Comment:
Perhaps we should add `boost::detail::winapi::FILE_FLAG_SEQUENTIAL_SCAN_`? It wouldn't become available until Boost 1.66.0 but can get it in now, see: https://github.com/boostorg/winapi/tree/develop/include/boost/detail/winapi

Reply:
Both `FILE_FLAG_RANDOM_ACCESS_` and `FILE_FLAG_SEQUENTIAL_SCAN_`?

Not entirely sure what you mean by getting them now.
Copy the headers into Beast? Copy just those definitions? Require consumers to get the pre-release headers?